### PR TITLE
add: Dockerfiles for integrated development environments

### PR DESCRIPTION
**Pull request by zurdhva**
**Dockerfile** that adds the packages required for each plug-in development, based on the environment in which **jdk**, **maven**, and **jenkins** are available.

**ikos** referred to the Docker file at [link](https://github.com/NASA-SW-VnV/ikos/blob/master/test/install/ubuntu-18.04/Dockerfile).

**Jenkins** also referred to the Docker file of the official repo.

--------------------------------------------------------------------------------------------------------------------------

Originally, Jenkins was installed in Ubuntu with the following command.
`wget -O - https://pkg.jenkins.io/debian/jenkins.io.key | sudo apt-key add `
`sudo sh -c 'echo deb https://pkg.jenkins.io/debian-stable binary/ > /etc/apt/sources.list.d/jenkins.list' `
`sudo apt-get update`

However, after the same process in the docker file, the Jenkins server cannot be opened when connecting to the container, so the file was longer because I referred to the Jenkins official repo docker file as above.

### CODE REVIEW

**Comment by iamwonseok** [comment]
@juheesvt , I met the error on the docker build command in Step 27/54 after I ran "docker build - < Dockerfile" using the PR. 
Please let me know if I should do something before executing docker build command.

File: Docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+# Refresh cache
+RUN apt-get update
+
+########## Jenkins Dockerfile ##########
+
+RUN apt-get upgrade -y \
+  && apt-get install -y --no-install-recommends git curl fontconfig unzip vim sudo clang-format-10\
+  && apt install -y maven \
+  && curl -s https://packagecloud.io/install/repositories/github/git-lfs/script.deb.sh | bash && apt-get install -y git-lfs && git lfs install \
```
Comment (iamwonseok) [inline Docker/Dockerfile]: Please remove the backslash character at the end of L36. It leads to "Empty continuation lines".

File: Docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+ARG REF=/usr/share/jenkins/ref
+
+ENV JENKINS_HOME $JENKINS_HOME
+ENV JENKINS_SLAVE_AGENT_PORT ${agent_port}
+ENV REF $REF
+
+# Jenkins is run with user `jenkins`, uid = 1000
+# If you bind mount a volume from the host or a data container,
+# ensure you use the same uid
+RUN mkdir -p $JENKINS_HOME
```
Comment (iamwonseok) [inline Docker/Dockerfile]: Use WORKDIR instead of RUN mkdir -p.

File: Docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+#  && useradd -d "$JENKINS_HOME" -u ${uid} -g ${gid} -m -s /bin/bash ${user}
+
+# Jenkins home directory is a volume, so configuration and build history
+# can be persisted and survive image upgrades
+VOLUME $JENKINS_HOME
+
+# $REF (defaults to `/usr/share/jenkins/ref/`) contains all reference configuration we want
+# to set on a fresh new installation. Use it to bundle additional plugins
+# or config file with your custom jenkins Docker image.
+RUN mkdir -p ${REF}/init.groovy.d
```
Comment (iamwonseok) [inline Docker/Dockerfile]: Ditto.

**Comment by zurdhva** [comment]
@iamwonseok 

I modified the above three points and pushed them after completing docker build and run test.
Thank you for your review!

File: Docker/Dockerfile (hunk 2 above)
Comment (zurdhva) [reply Docker/Dockerfile]: I modified it as you said, is there any special reason to use `WORKDIR` instead of `mkdir -p`?

Comment (iamwonseok) [reply Docker/Dockerfile]: Using mkdir, sometimes it fails to make directory if the directory exists.

**Comment by iamwonseok** [comment]
@juheesvt

I am still suffering the error from Step 28/56.

Step 28/56 : COPY tini_pub.gpg ${JENKINS_HOME}/tini_pub.gpg
COPY failed: stat /var/lib/docker/tmp/docker-builder536964064/tini_pub.gpg: no such file or directory

I run "docker build - < Dockerfile" on ubuntu 20.04 LTS.

**Comment by zurdhva** [comment]
@iamwonseok 

If I build a docker file using the command `docker build -t jenkins-test .`, it will succeed and connect to the container well, but if I build it using the `docker build - < Dockerfile` you wrote, it will fail in the same part.

May I ask you the difference between the two orders?

**Comment by zurdhva** [comment]
First of all, the COPY command doesn't seem to be eating well.

**Comment by iamwonseok** [comment]
@juheesvt, could you please add how to build your docker image in README file?

**Comment by iamwonseok** [comment]
Overall looks good to me :D

**Pull request merged**